mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -76,60 +74,45 @@ def infer_and_parse_json_input(json_input, schema: Schema = None):
                        DataFrame, or a stream containing such a string representation.
     :param schema: Optional schema specification to be used during parsing.
     """
-    try:
-        decoded_input = json.loads(json_input)
-    except json.decoder.JSONDecodeError:
-        _handle_serving_error(
-            error_message=(
-                "Failed to parse input from JSON. Ensure that input is a valid JSON"
-                " formatted string."
-            ),
-            error_code=BAD_REQUEST,
-        )
-
-    if isinstance(decoded_input, list):
-        return parse_json_input(json_input=json_input, orient="records", schema=schema)
-    elif isinstance(decoded_input, dict):
-        if "instances" in decoded_input or "inputs" in decoded_input:
-            try:
-                return parse_tf_serving_input(decoded_input, schema=schema)
-            except MlflowException as ex:
-                _handle_serving_error(
-                    error_message=(ex.message),
-                    error_code=BAD_REQUEST,
-                )
-        else:
-            return parse_json_input(json_input=json_input, orient="split", schema=schema)
+    if isinstance(json_input, dict):
+        decoded_input = json_input
     else:
-        _handle_serving_error(
-            error_message=(
-                "Failed to parse input from JSON. Ensure that input is a valid JSON"
-                " list or dictionary."
-            ),
-            error_code=BAD_REQUEST,
+        try:
+            decoded_input = json.loads(json_input)
+        except json.decoder.JSONDecodeError as ex:
+            raise MlflowException(
+                message=(
+                    "Failed to parse input from JSON. Ensure that input is a valid JSON"
+                    f" formatted string. Error: '{ex}'"
+                ),
+                error_code=BAD_REQUEST,
+            )
+    if isinstance(decoded_input, dict):
+        format_keys = set(decoded_input.keys()).intersection(SUPPORTED_FORMATS)
+        if len(format_keys) != 1:
+            message = f"Received dictionary with input fields: {format_keys}"
+            raise MlflowException(
+                message=f"{REQUIRED_INPUT_FORMAT}. {message}.", error_code=BAD_REQUEST
```

Comment:
Nit: `REQUIRED_INPUT_FORMAT` already contains a period. Messages containing `{REQUIRED_INPUT_FORMAT}.` will contain two periods - `..`

Reply:
nice catch.

Reply:
@dbczumar Reminder to address this nit of a comment :)

Reply:
Nevermind, already seems to be addressed